flytekit/models/sagemaker/parameter_ranges.py

```diff
@@ -0,0 +1,237 @@
+        :return: _idl_parameter_ranges.IntegerParameterRange
+        """
+
+        if self.scaling_type == _sdk_sagemaker_types.HyperparameterScalingType.AUTO:
+            scaling_type = _idl_parameter_ranges.HyperparameterScalingType.AUTO
+        elif self.scaling_type == _sdk_sagemaker_types.HyperparameterScalingType.LINEAR:
+            scaling_type = _idl_parameter_ranges.HyperparameterScalingType.LINEAR
+        elif self.scaling_type == _sdk_sagemaker_types.HyperparameterScalingType.LOGARITHMIC:
+            scaling_type = _idl_parameter_ranges.HyperparameterScalingType.LOGARITHMIC
+        elif self.scaling_type == _sdk_sagemaker_types.HyperparameterScalingType.REVERSELOGARITHMIC:
+            scaling_type = _idl_parameter_ranges.HyperparameterScalingType.REVERSELOGARITHMIC
+        else:
+            raise _user_exceptions.FlyteValidationException("Invalid SageMaker HyperparameterScalingType specified")
+
+        return _idl_parameter_ranges.IntegerParameterRange(
+            max_value=self._max_value,
+            min_value=self._min_value,
+            scaling_type=scaling_type,
+        )
+
+    @classmethod
+    def from_flyte_idl(cls, pb2_object):
+
+        scaling_type = _sdk_sagemaker_types.HyperparameterScalingType.AUTO
+        if pb2_object.type == _idl_parameter_ranges.HyperparameterScalingType.LINEAR:
+            scaling_type = _sdk_sagemaker_types.HyperparameterScalingType.LINEAR
+        elif pb2_object.type == _idl_parameter_ranges.HyperparameterScalingType.LOGARITHMIC:
+            scaling_type = _sdk_sagemaker_types.HyperparameterScalingType.LOGARITHMIC
+        elif pb2_object.type == _idl_parameter_ranges.HyperparameterScalingType.REVERSELOGARITHMIC:
+            scaling_type = _sdk_sagemaker_types.HyperparameterScalingType.REVERSELOGARITHMIC
+
+        return cls(
+            max_value=pb2_object.max_value,
+            min_value=pb2_object.min_value,
+            scaling_type=scaling_type,
+        )
+
+
+class CategoricalParameterRange(_common.FlyteIdlEntity):
+    def __init__(
+            self,
+            values: List[str],
+    ):
+        """
+
+        :param List[str] values: list of strings representing categorical values
+        """
+        self._values = values
+
+    @property
+    def values(self):
+        """
+        :return: list[string]
+        """
+        return self._values
+
+    def to_flyte_idl(self):
+        """
+        :return: _idl_parameter_ranges.CategoricalParameterRange
+        """
+        return _idl_parameter_ranges.CategoricalParameterRange(
+            values=self._values
+        )
+
+    @classmethod
+    def from_flyte_idl(cls, pb2_object):
+        return cls(
+            values=pb2_object.values
+        )
+
+
+class ParameterRanges(_common.FlyteIdlEntity):
+    def __init__(
+            self,
+            parameter_range_map: Dict[str, _common.FlyteIdlEntity],
+    ):
+        self._parameter_range_map = parameter_range_map
+
+    def to_flyte_idl(self):
```

Comment:
add type hint or type comment

Reply:
If you `from __future__ import annotations` at the top of the file, you can do forward references even in python 3.6.  That is, you can use the class as a type hint in the class definition itself.

See https://www.python.org/dev/peps/pep-0563/ for more info.

Reply:
Since our CI is using 3.6.3 and `from __future__ import annotations` is only supported starting python 3.7, I'll remove this import and remove the type hints for `self`s. Thanks for the suggestion tho.